src/cryptography/hazmat/backends/openssl/ec.py

```diff
@@ -275,6 +275,28 @@ def public_numbers(self):
             curve=self._curve
         )
 
+    def encode_point(self, compressed):
```

Comment:
No real argument beyond the probable need to switch to a whitelisting scheme at all our callsites so that invalid combinations remain impossible. If we're willing to expand `public_bytes` then we do need a name for these formats. `PublicFormat.UncompressedPoint` and `PublicFormat.CompressedPoint`?

Reply:
And what encoding do we need for those formats then? This isn't PEM or DER, but "raw" isn't really accurate either.

Reply:
I think those names are reasonable. Why do you think Raw isn't accurate?

FWIW, you didn't really answer my question, why this PR took the opposite approach of #4653?

Reply:
It is binary, but it contains a prefix that denotes the type (x04 for uncompressed points, x02 and x03 for compressed). If it it has structure like that it feels wrong to call it raw to me.

There is no reason other than to demonstrate two different approaches we could take. That's why both of these PRs were put up for discussion of the API and not review/merge worthy yet 😄 

I would prefer not to proliferate serialization methods so moving to purely through `public_bytes` is 👍 from me. However, we should be cognizant of the unfortunate situation we find ourselves in where we're adding values to the enums that have no meaning in the other methods. We're deriving little to no value from them being enums since we can't use isinstance checks and must instead check for the subset of allowable values from the enums.

Reply:
👍 ok, it wasn't totally clear to me that they were _intentionally_ opposite approaches :-)

We can call it SECGBlahBlahBlah instead of Raw.

I'm also not wild about the mess of these enums either, but I think fixing it requires redoing a bunch of our serialization API, and I'm not sure it's a huge gain.

Reply:
Is it SECG1 that defines the uncompressed point prefix? I didn't spot it in the text. ANSI X9.62 may also be relevant here but [this draft](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.202.2977&rep=rep1&type=pdf) (since we can't look at the final spec without $$$) doesn't appear to state that uncompressed points have prefix \x04

Reply:
Oh I'm a liar. X9.62 section 4.3.6.